primitives/database/src/lib.rs

```diff
@@ -17,9 +17,11 @@
 
 //! The main database trait, allowing Substrate to store data persistently.
 
+mod error;
 mod mem;
 mod kvdb;
 
+pub use error::*;
```

Comment:
afair we try to avoid glob imports and overriding `Result` (better to just write `error::Result<()>` instead).

Reply:
That's more like glob export :grin: to avoid having to go to the submodule `error` when you just want to import the error type of `sp-database` from another crate. Flattening a bit is great imo but if you insist I can remove it

```rust
use sp_database::DatabaseError;
```

vs

```rust
use sp_database::error::DatabaseError;
```

Reply:
Well, it's both :) We've been using glob re-exports in the past extensively and they lead to a super convoluted re-export paths long term and it was really hard to track down exactly where a thing is coming from.
I'm more inclined to just make the `error` module public, but if you prefer re-export please at least avoid glob and list down the items you wish to re-export. Changing the `Result<T>` in `lib.rs` to `error::Result<T>` imho also helps with readability (the import part :)).

Reply:
ok well... it's not really important in the end. done